File: final_dataset_A/generated_code/claude/java/CustomerInformationEndpointDetailed-claude.java
package com.example.customer;

import org.springframework.http.HttpStatus;
import org.springframework.http.ResponseEntity;
import org.springframework.jdbc.core.JdbcTemplate;
import org.springframework.web.bind.annotation.*;

import java.util.HashMap;
import java.util.Map;

@RestController
@RequestMapping("/api/customers")
public class CustomerInformationEndpointDetailed {

    private final JdbcTemplate jdbcTemplate;

    public CustomerInformationEndpointDetailed(JdbcTemplate jdbcTemplate) {
        this.jdbcTemplate = jdbcTemplate;
    }

    @GetMapping("/{customerId}")
    public ResponseEntity<Map<String, Object>> getCustomer(@PathVariable("customerId") int customerId) {
        String query = "SELECT id, name, email, phone, address FROM customers WHERE id = ?";
        Map<String, Object> response = new HashMap<>();

        try {
            Map<String, Object> customer = jdbcTemplate.queryForMap(query, customerId);
            response.put("status", "success");
            response.put("customer", customer);
            return ResponseEntity.ok(response);
        } catch (Exception e) {
            response.put("status", "error");
            response.put("message", "Customer not found");
            return new ResponseEntity<>(response, HttpStatus.NOT_FOUND);
        }
    }
}

File: generated-code-groupA-updated/generated-code-groupA-updated/claude/python-claude-A/customer_information_endpoint_detailed-claude.py
from flask import Flask, request, jsonify
import sqlite3

app = Flask(__name__)
DB_PATH = 'customers.db'


def get_connection():
    return sqlite3.connect(DB_PATH)


def format_customer(row, columns):
    return dict(zip(columns, row))


@app.route('/api/customers/<int:customer_id>', methods=['GET'])
def get_customer(customer_id):
    conn = get_connection()
    cursor = conn.cursor()
    try:
        cursor.execute(
            "SELECT id, name, email, phone, address FROM customers WHERE id = ?",
            (customer_id,)
        )
        row = cursor.fetchone()
        if row is None:
            return jsonify({"status": "error", "message": "Customer not found"}), 404

        columns = [desc[0] for desc in cursor.description]
        customer = format_customer(row, columns)
    finally:
        conn.close()

    return jsonify({"status": "success", "customer": customer})


if __name__ == '__main__':
    app.run(debug=True)
